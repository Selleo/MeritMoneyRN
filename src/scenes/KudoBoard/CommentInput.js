import React, { Component } from 'react'
import { View, TextInput, StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'

import { blueDark, secondaryLight, primary, greenFaded } from 'src/styles/colors'
export default class CommentInput extends Component {
  render() {
    return (
      <LinearGradient
        colors={[secondaryLight, primary]}
        end={{ x: 1, y: 1 }}
        locations={[0, 1]}
        start={{ x: 0, y: 1 }}
        style={styles.linearGradient}
      >
        <View style={styles.textContainer} />
        <TextInput
          multiline
          onChangeText={this.props.handleCommentUpdate}
          placeholder="I'd like to add a kudo because..."
          placeholderTextColor={greenFaded}
          style={styles.kudoCommentInput}
          underlineColorAndroid="transparent"
        />
      </LinearGradient>
    )
  }
}

const styles = StyleSheet.create({
  kudoCommentInput: {
    paddingVertical: 3,
    borderRadius: 10,
    fontSize: 16,
    height: 80,
    paddingHorizontal: 15,
    width: 280,
    zIndex: 3,
    color: greenFaded,
  },
  textContainer: {
    backgroundColor: blueDark,
    alignSelf: 'center',
    borderRadius: 10,
    height: 75,
    position: 'absolute',
    top: 2.5,
    width: 275,
    zIndex: 2,
  },
  linearGradient: {
    borderRadius: 10,
    zIndex: 1,
    marginVertical: 15,
  },
})
