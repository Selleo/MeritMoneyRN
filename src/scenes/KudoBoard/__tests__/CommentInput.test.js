import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import CommentInput from '../CommentInput'

describe('<CommentInput />', () => {
  const defaultProps = {
    handleChange: jest.fn(),
  }

  const wrapper = renderer.create(<CommentInput {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
