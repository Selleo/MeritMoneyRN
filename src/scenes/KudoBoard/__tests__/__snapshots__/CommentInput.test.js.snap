// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentInput /> render 1`] = `
<BVLinearGradient
  colors={
    Array [
      4282147443,
      4281317757,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  style={
    Object {
      "borderRadius": 10,
      "marginVertical": 15,
      "zIndex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#222f3f",
        "borderRadius": 10,
        "height": 75,
        "position": "absolute",
        "top": 2.5,
        "width": 275,
        "zIndex": 2,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[MockFunction]}
    placeholder="I'd like to add a kudo because..."
    placeholderTextColor="#7f9d90"
    style={
      Object {
        "borderRadius": 10,
        "color": "#7f9d90",
        "fontSize": 16,
        "height": 80,
        "paddingHorizontal": 15,
        "paddingVertical": 3,
        "width": 280,
        "zIndex": 3,
      }
    }
    underlineColorAndroid="transparent"
  />
</BVLinearGradient>
`;
